airbyte-cdk/python/unit_tests/sources/file_based/scenarios/scenario_builder.py

```diff
@@ -90,110 +90,112 @@ def configured_catalog(self, sync_mode: SyncMode) -> Optional[Dict[str, Any]]:
 
         return catalog
 
-    def input_state(self) -> List[Dict[str, Any]]:
+    def input_state(self) -> List[Mapping[str, Any]]:
         if self.incremental_scenario_config:
             return self.incremental_scenario_config.input_state
         else:
             return []
 
 
 class TestScenarioBuilder:
-    def __init__(self):
+    def __init__(self) -> None:
         self._name = ""
-        self._config = {}
-        self._files = {}
-        self._file_type = None
-        self._expected_check_status = None
-        self._expected_catalog = {}
-        self._expected_logs = {}
-        self._expected_records = {}
-        self._availability_strategy = None
-        self._discovery_policy = DefaultDiscoveryPolicy()
-        self._validation_policies = None
+        self._config: Mapping[str, Any] = {}
+        self._files: Mapping[str, Any] = {}
+        self._file_type: Optional[str] = None
+        self._expected_check_status: Optional[str] = None
+        self._expected_catalog: Mapping[str, Any] = {}
+        self._expected_logs: List[Mapping[str, Any]] = []
+        self._expected_records: List[Mapping[str, Any]] = []
+        self._availability_strategy: Optional[AvailabilityStrategy] = None
+        self._discovery_policy: AbstractDiscoveryPolicy = DefaultDiscoveryPolicy()
+        self._validation_policies: Optional[Mapping[str, AbstractSchemaValidationPolicy]] = None
         self._parsers = default_parsers
-        self._stream_reader = None
-        self._expected_check_error = None, None
-        self._expected_discover_error = None, None
-        self._expected_read_error = None, None
-        self._incremental_scenario_config = None
-        self._file_write_options = {}
-
-    def set_name(self, name: str):
+        self._stream_reader: Optional[AbstractFileBasedStreamReader] = None
+        self._expected_check_error: Tuple[Optional[Type[Exception]], Optional[str]] = None, None
+        self._expected_discover_error: Tuple[Optional[Type[Exception]], Optional[str]] = None, None
+        self._expected_read_error: Tuple[Optional[Type[Exception]], Optional[str]] = None, None
+        self._incremental_scenario_config: Optional[IncrementalScenarioConfig] = None
+        self._file_write_options: Mapping[str, Any] = {}
+
+    def set_name(self, name: str) -> "TestScenarioBuilder":
```

Comment:
Why string literals here for the builder methods?

Reply:
because the type doesn't exist when we hit that line.

From [pep-0484](https://peps.python.org/pep-0484/#the-problem-of-forward-declarations):
> Our solution, which isn’t particularly elegant, but gets the job done, is to allow using string literals in annotations